### DISCUSSION TITLE
A quick Relative JSON Pointer release

### DISCUSSION THREAD

**Discussion by handrews** [json-schema-org/community#225] [Specification]
Hi folks,
  I'd like to do a quick update of the Relative JSON Pointer (RJP) draft (as an IETF I-D before we move away from that).  I am happy to do all of the work involved, I just want to make sure it will not be blocked before I start on it, and that someone is available to review/discuss the one non-trivial thing involved.

There is very little to do, but one thing is quite important:

* json-schema-org/json-schema-spec#1121 is more weird phrasing than actual bug, but it already has PR json-schema-org/json-schema-spec#1147
* json-schema-org/json-schema-spec#1175 is a trivial fix to align the ABNF with the specification text.
* the other two issues are alternate proposals for the same problem, which is the main one I want to fix:  it is not clear how to handle certain cases of resolving a Relative JSON Pointer against a (regular a.k.a. absolute) JSON Pointer, particularly when `-` (the index beyond the last existing index of an array, because the current text assumes existing locations) or `#` (the operator to take the name/index of the property/array element, which is not part of normal JSON Pointer syntax since the name/index is always present in the pointer text in that case) or both are involved
    * json-schema-org/json-schema-spec#1235 proposes a process that need not involve evaluating the regular JSON Pointer first
    * json-schema-org/json-schema-spec#1236 proposes sticking with instance-based resolution, but allowing certain non-existent instance locations to be tolerated during the process

Relative JSON Pointers are important for implementing templated annotations (like Hyper-Schema's `links`), and currently some very reasonable uses of RJP are under-specified and can't really be implemented.

**Comment by jdesrosiers** [comment]
I'm not involved with the RJP spec, but I just wanted to say that just because JSON Schema wants to move away from using I-Ds, this doesn't need to be the case for RJP. I think RJP would make sense under the IETF process if that's what you want to do. I'm not involved in that spec, so I won't express an opinion. I'm just saying it's a reasonable option.

**Reply by handrews** [reply]
Good point.  I figure at minimum there's no need to change now for a quick update to plug a gap in the spec and fix wording bugs.

**Comment by handrews** [comment]
Due to a lack of anything resembling an objection, I'm going to move ahead with this.  Please comment with any concerns, though.
